internal/commands/secrets/create.go

```diff
@@ -0,0 +1,58 @@
+package secrets
+
+import (
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+
+	"github.com/spf13/pflag"
+)
+
+// CommandCreate is the `secrets create` command
+type CommandCreate struct {
+	inputs      createInputs
+	realmClient realm.Client
+}
+
+// Flags is the command flags
+func (cmd *CommandCreate) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.StringVarP(&cmd.inputs.Name, flagName, flagNameShort, "", flagNameUsage)
+	fs.StringVarP(&cmd.inputs.Value, flagValue, flagValueShort, "", flagValueUsage)
+}
+
+// Inputs is the command inputs
+func (cmd *CommandCreate) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Setup is the command setup
+func (cmd *CommandCreate) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandCreate) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := app.Resolve(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+
+	if app.ID == "" {
```

Comment:
what scenario is this guarding against?  when could we resolve an app without an id?

Reply:
That's correct, I removed